inferno-core/src/Inferno/Eval.hs

```diff
@@ -204,6 +193,16 @@ eval env@(localEnv, pinnedEnv) expr = case expr of
       _ -> throwM $ RuntimeError "failed to match with a bool"
   Tuple_ es ->
     foldrM (\(e, _) vs -> eval env e >>= return . (: vs)) [] (tListToList es) >>= return . VTuple
+  Record_ fs -> do
+    valMap <- foldrM (\(f, e, _) vs -> eval env e >>= \v -> return ((f, v) : vs)) [] fs
+    return $ VRecord $ Map.fromList valMap
+  RecordField_ (Ident r) f -> do
+    eval env (Var undefined Nothing LocalScope $ Expl $ ExtIdent $ Right r) >>= \case
```

Comment:
I guess this `undefined` is safe, but maybe a short comment explaining why?

Reply:
An `error` with a searchable message also works and if the impossible happens one has something to search.

Reply:
Could alternatively just inline the eval call rhs:

```haskell
case Map.lookup (ExtIdent $ Right r) localEnv of
      Just v -> return v
Nothing -> throwM $ RuntimeError $ show (ExtIdent $ Right r) <> " not found in the unpinned env"
```